server/controller/src/main/java/ai/starwhale/mlops/common/TarFileUtil.java

```diff
@@ -101,6 +101,21 @@ public static byte[] getContent(InputStream is) {
 
     }
 
+    public static void extract(
+            InputStream in, ArchiveFileConsumer func) throws ArchiveException {
+        try (ArchiveInputStream archive = getArchiveInputStream(in)) {
+            TarArchiveEntry entry;
+            while ((entry = (TarArchiveEntry) archive.getNextEntry()) != null) {
+                if (entry.isDirectory()) {
+                    continue;
```

Comment:
All directories will be ignored. Try to process the entry recursively?

Reply:
the entries are like: path/, path/model.pth,this will only ignore 'path/'（there is no need to request a dir put operation）,  so the original directory will be preserved

Reply:
is this a model specific util or a common tar util?